src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -359,6 +372,17 @@ public static PreferenceSettings Load(string filePath)
             settings.CustomPackageFolders = settings.CustomPackageFolders.Distinct().ToList();
 
             return settings;
+		}
+
+        internal void InitializeNamespacesToExcludeFromLibrary()
+        {
+            if (!NamespacesToExcludeFromLibrarySpecified)
+            {
+                NamespacesToExcludeFromLibrary.Add(
+                    "ProtoGeometry.dll:Autodesk.DesignScript.Geometry.TSpline"
```

Comment:
We need to distinguish between an "empty XML element" (no namespace to exclude, hence show T-Spline nodes) and a "missing XML element" (this version of Dynamo has never been run, hence hide T-Spline nodes).

In the "missing element" case, normal deserialization will still return an empty list, but this boolean property is false.